Re-adding the Methanol Byproduct

Somus88 opened the pull request:
Re-adding Methanol byproduct, also with the fixed value.

Related, Issue number #53836

<!-- HOW TO USE: Under each "#### Heading" below, enter information relevant to your pull request.
Leave the headings unless they don't apply to your PR.

Please read carefully and don't delete the comments delimited by "< !--" and "-- >"
Once a pull request is submitted automatic stylistic and consistency checks will be performed on the PR's changes.
The results of these can be either seen under the "Files changed" section of a PR or in the check's details.

NOTE: Please grant permission for repository maintainers to edit your PR.  It is EXTREMELY common for PRs to be held up due to trivial changes being requested and the author being unavailable to make them. -->

#### Summary
Content "Re-adding the Methanol Byproduct from Crafting Moonshine Mash"
<!-- This section should consist of exactly one line, edit the one above.
Category must be one of these: Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N. Or replace the whole line with just the word None for no changelog entry.
For more on the meaning of each category, see:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md
If approved and merged, your summary will be added to the project changelog:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt -->

#### Purpose of change
To fix the methanol byproduct amount. (creates 68L of Methanol from water, pumpkin, sugar, and yeast.)
Should Fix #53836 
Steps to Reproduce:

1. Make a world, spawn in 15 units of water in any container, 12 Pumpkin, 50 Sugar, 2 yeast, and the 'family cookbook' book for the recipe.
2. Craft a single batch of moonshine mash.
3. Watch as the game gives you 68L of Methanol.
<!-- With a few sentences, describe your reasons for making this change.  If it relates to an existing issue, you can link it with a # followed by the GitHub issue number, like #1234.  If your pull request *fully* resolves an issue, include the word "Fix" or "Fixes" before the issue number, like: Fixes #1234.

If there is no related issue, explain here what issue, feature, or other concern you are addressing.  If this is a bugfix, include steps to reproduce the original bug, so your fix can be verified. -->

#### Describe the solution
The solution would be to re-add the byproduct and fix the amount produced to 2 units of methanol, which will result in 500mL of methanol, which is around 13-14% of the resulting product.
<!-- How does the feature work, or how does this fix a bug?  The easier you make your solution to understand, the faster it can get merged. -->

#### Describe alternatives you've considered
Removal of methanol byproduct entirely in the books however I personally would still like cheap quick methanol :)
<!-- Explain any alternative solutions, different approaches, or possibilities you've considered using to solve the same problem. -->

#### Testing

None because I don't know how to + it's pretty trivial

<!-- Describe what steps you took to test that this PR resolved the bug or added the feature, and what tests you performed to make sure it didn't cause any regressions.  Also include testing suggestions for reviewers and maintainers. -->

Alex-Folts commented:
I see 2 problems with this:

- it is unfermented stuff, basically it is just all ingredients mixed together - at this point there is no ethanol and methanol in it, it need to stay for some time to generate it(fermentation need to occur). Means you cant get out methanol at this point.
- 500ml of methanol is too much for about 4L of initial stuff, there is methanol and aceton and some other volatiles, but not of this quantity. Those volatiles people call heads, they come first when you distilling and there not much of them, need to google some numbers but should be much less than this. But again, it will matter on later stage when you distilling **fermented** stuff.

Somus88 commented:
> I see 2 problems with this:
> 
> * it is unfermented stuff, basically it is just all ingredients mixed together - at this point there is no ethanol and methanol in it, it need to stay for some time to generate it(fermentation need to occur). Means you cant get out methanol at this point.
> * 500ml of methanol is too much for about 4L of initial stuff, there is methanol and aceton and some other volatiles, but not of this quantity. Those volatiles people call heads, they come first when you distilling and there not much of them, need to google some numbers but should be much less than this. But again, it will matter on later stage when you distilling **fermented** stuff.

I have changed where Methanol is produced to the distillation process, and fixed the amount accordingly to Wikipedia (it says 50-150mL of foreshot that contains most of methanol,) unfortunately most other websites I found are rather questionable in terms of reliability.

Maleclypse commented on data/json/recipes/food/distill.json:
```diff
@@ -81,8 +81,7 @@
     "result": "moonshine",
-    "//": "should produce 12-15% foreshot with significant methanol",
-    "//2": "1 unit of methanol in the JSON is 250 units in game",
+    "byproducts": [ [ "chem_methanol", 1 ] ],
```
```suggestion
    "byproducts": [ [ "chem_methanol", 1 ] ],
    "//": "kick test",
```

Maleclypse commented on data/json/recipes/food/distill.json:
```diff
@@ -81,8 +81,8 @@
-    "//": "should produce 12-15% foreshot with significant methanol",
-    "//2": "1 unit of methanol in the JSON is 250 units in game",
+    "byproducts": [ [ "chem_methanol", 1 ] ],
+    "//": "kick test",
```
```suggestion
```
